neural_compressor/adaptor/torch_utils/gptq.py

```diff
@@ -0,0 +1,586 @@
+import math
+import time
+
+import torch
+import torch.nn as nn
+import transformers
+from tqdm import tqdm
+from functools import partial
+# different models may have different input structures. 
+# arch_inputs = {
+#     'opt': ['attention_mask'],
+#     'bloom': ['attention_mask', 'alibi'],
+#     'llama': ['attention_mask', 'position_ids']
+# }
+
+DEBUG = False 
+
+def is_leaf(module):
+    children_cnt = 0
+    for n in module.children():
+        children_cnt += 1
+    return True if children_cnt == 0 else False
+
+def trace_gptq_target_blocks(module, module_types = [torch.nn.ModuleList]):
+    """
+    seperately trace two parts related to gptq:
+    1. embedding layers (layers before transformer stacks)
+    2. transformer stacks
+    """
+    gptq_related_blocks = {
+        "embeddings": {},
+        "transformers_pre": {}, # todo
+        "transformers_name": None,
+        "transformers": None,

```

Comment:
As GPTQ only handles transformer blocks, there are some layers besides the last layer that not be quantized, e.g. opt-350m. We need to let the users or kernel team know which layer has been quantized.

Reply:
please dump the unquantized layer name to a log file or better sync with Hexin how to export this information.